src/testdir/test_quickfix.vim

```diff
@@ -5281,7 +5308,7 @@ func Test_add_invalid_entry_with_qf_window()
   call setqflist(['bb'], 'a')
   call assert_equal(1, line('$'))
   call assert_equal(['Xfile1|10| aa'], getline(1, '$'))
-  call assert_equal([{'lnum': 10, 'bufnr': bufnr('Xfile1'), 'col': 0, 'pattern': '', 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'module': '', 'text': 'aa'}], getqflist())
+  call assert_equal([{'lnum': 10, 'end_lnum': 0, 'bufnr': bufnr('Xfile1'), 'col': 0, 'end_col': 0, 'pattern': '', 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'module': '', 'text': 'aa'}], getqflist())
```

Comment:
what if 'end_lnum' and/or 'end_col' is '0' (or even was a negative value), then what's the `getline(1, '$')` ?
would it be still compatible like `Xfile1|10| aa` vs not `Xfile1|10-0| aa` ? the later case should be not wanted.

Reply:
Don't worry.
`end_lnum` and `end_col` are just optional values. The text in the quickfix window is never changed when they are zero. It will be displayed as before.  See [the comment above](https://github.com/vim/vim/pull/8393#issuecomment-863393256).

Reply:
ok, wish so.
i mean let it be part of that test too. :-)